src/main/java/it/pagopa/transactions/commands/handlers/TransactionRequestAuthorizationHandler.java

```diff
@@ -47,31 +53,42 @@ public Mono<RequestAuthorizationResponseDto> handle(TransactionRequestAuthorizat
             return Mono.error(new AlreadyProcessedException(transaction.getRptId()));
         }
 
-        return paymentGatewayClient.requestAuthorization(command.getData())
-                .flatMap(gatewayResponse -> {
+        return paymentGatewayClient.requestGeneralAuthorization(command.getData())
+                .flatMap(authResponse -> {
                     log.info("Logging authorization event for rpt id {}", transaction.getRptId().value());
 
-                    TransactionAuthorizationRequestedEvent authorizationEvent = new TransactionAuthorizationRequestedEvent(
-                            transaction.getTransactionId().value().toString(),
-                            transaction.getRptId().value(),
-                            transaction.getTransactionActivatedData().getPaymentToken(),
-                            new TransactionAuthorizationRequestData(
-                                    command.getData().transaction().getAmount().value(),
-                                    command.getData().fee(),
-                                    command.getData().paymentInstrumentId(),
-                                    command.getData().pspId(),
-                                    command.getData().paymentTypeCode(),
-                                    command.getData().brokerName(),
-                                    command.getData().pspChannelCode(),
-                                    command.getData().paymentMethodName(),
-                                    command.getData().pspBusinessName(),
-                                    gatewayResponse.getRequestId()));
+                    Mono<Tuple2<String,String>> monoPostePay = Mono.just(authResponse.getT1()).filter(pPayAuth -> pPayAuth.isPresent())
+                            .switchIfEmpty(Mono.empty())
+                            .flatMap(pPay -> Mono.zip(Mono.just(pPay.get().getRequestId()),Mono.just(pPay.get().getUrlRedirect())));
 
-                    return transactionEventStoreRepository.save(authorizationEvent)
-                            .thenReturn(gatewayResponse)
-                            .map(auth -> new RequestAuthorizationResponseDto()
-                                    .authorizationUrl(auth.getUrlRedirect())
-                                    .authorizationRequestId(auth.getRequestId()));
+                    Mono<Tuple2<String,String>> monoXPay = Mono.just(authResponse.getT2()).filter(xPayAuth -> xPayAuth.isPresent())
+                            .switchIfEmpty(Mono.empty())
+                            .flatMap(xPay -> Mono.zip(Mono.just(xPay.get().getRequestId()),Mono.just(xPay.get().getUrlRedirect())));
+
+                    return monoPostePay.switchIfEmpty(monoXPay)
```

Comment:
regardless of payment method, in response we will always get an `idRequest` and `urlRedirect`, could we generalize above so as to avoid this double check?

Reply:
Do you think that a data structure (dto, map, list) to handle all payment gateway response could be a fine solution?